Assets/_BForBoss/_Core/Scripts/Environment/ItemPickupBehaviour.cs

```diff
@@ -27,7 +27,7 @@ public void CleanUp()
         
         private void OnTriggerEnter(Collider other)
         {
-            if (other.TryGetComponent(out PlayerLifeCycleBehaviour _))
+            if (other.TryGetComponent(out PlayerLifeCycleBehaviour playerLife) && !playerLife.IsFullHealth)
```

Comment:
This would mean all items (not just items that heal you supposedly) would not work if Player Health is 100% right?

Not sure that seems like the correct behaviour.

Reply:
An idea would be to have `HealingBehaviour` derive from something like `ItemBehaviour` in which it has a virtual method i.e. `public virtual bool CanPickUp()`.

and instead `_itemToPickUp` being a `GameObject` it would be a `ItemBehaviour`. Thus, this check would be 

`if (other.TryGetComponent(out PlayerLifeCycleBehaviour playerLife) && _itemToPickup.CanPickUp())`

What do you think?

Reply:
>This would mean all items (not just items that heal you supposedly) would not work if Player Health is 100% right?

that's the point of the ticket, that it won't work when it's 100% or more.

If it's another type of pickup item, it won't enter the if statement and just go to the next statement. E.g.

```
if (TryGet && IsFullHealth) {
    doSomething();
}

if (TryGet && SomethingElse) {
    doSomethingElse();
}
```

Reply:
>Not sure that seems like the correct behaviour.

You can Try it out in game and see if u can find any edge cases that won't work? 
Basically if it's <100%, it will work.